api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java

```diff
@@ -170,17 +175,74 @@ void shouldSaveClaimantWithClaimStatus(EligibilityStatus eligibilityStatus) {
     }
 
     @Test
-    void shouldSaveDuplicateClaimantForMatchingNino() {
+    void shouldUpdateClaimForMatchingNinoWhenEligible() {
         //given
-        Claimant claimant = aValidClaimant();
-        given(eligibilityAndEntitlementService.evaluateNewClaimant(any()))
-                .willReturn(EligibilityAndEntitlementDecision.buildWithStatus(DUPLICATE));
+        Claimant existingClaimant = aClaimantWithExpectedDeliveryDate(null);
+        LocalDate expectedDeliveryDate = LocalDate.now().plusMonths(6);
+        Claimant newClaimant = aClaimantWithExpectedDeliveryDate(expectedDeliveryDate);
+        Claim existingClaim = aClaimWithClaimant(existingClaimant);
+        UUID existingClaimId = UUID.randomUUID();
+        given(eligibilityAndEntitlementService.evaluateNewClaimant(any())).willReturn(EligibilityAndEntitlementDecision.builder()
+                .eligibilityStatus(ELIGIBLE)
+                .existingClaimId(existingClaimId)
+                .voucherEntitlement(aPaymentCycleVoucherEntitlementWithVouchers())
+                .build());
+        given(claimRepository.findById(any())).willReturn(Optional.of(existingClaim));
 
         //when
-        ClaimResult result = newClaimService.createClaim(claimant);
+        ClaimResult result = newClaimService.createOrUpdateClaim(newClaimant);
 
         //then
-        verify(eligibilityAndEntitlementService).evaluateNewClaimant(claimant);
+        assertThat(result.getClaim()).isEqualTo(existingClaim);
+        assertThat(result.getClaim().getClaimant()).isEqualTo(existingClaimant);
```

Comment:
To make it a bit clearer that this is matching by id, could we just compare the ids here?

Reply:
Updated to just check the claim and eed which I think is a bit clearer.